datafu-spark/src/main/resources/pyspark_utils/df_utils.py

```diff
@@ -0,0 +1,171 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+ 
+import pyspark
+from pyspark.sql import DataFrame
+from pyspark_utils.bridge_utils import _getjvm_class
+
+
+def _get_utils(df):
+    _gateway = df._sc._gateway
+    return _getjvm_class(_gateway, "datafu.spark.SparkDFUtilsBridge")
+
+
+# public:
+
+
+def dedup(df, group_col, order_cols = []):
```

Comment:
I think dedup/dedup2 should have semantic names. Make it clear what they do.

Reply:
changed names of `dedup` and `dedup2` to `dedup_with_order` and `dedup_with_combiner`